sdk/src/lib.rs

```diff
@@ -109,6 +117,100 @@ impl SP1Prover {
+        callback_datas: Vec<&str>,
+    ) -> Result<Vec<String>> {
+        let client = NetworkClient::with_token(access_token);
+        let verifier = SP1_VERIFIER_ADDRESS;
+
+        let mut tx_details = Vec::new();
+        for ((i, &callback), &callback_data) in
+            callbacks.iter().enumerate().zip(callback_datas.iter())
+        {
+            if let Some(&chain_id) = chain_ids.get(i) {
+                let tx_id = client
+                    .relay_proof(proof_id, chain_id, verifier, callback, callback_data)
+                    .await
+                    .with_context(|| format!("Failed to relay proof to chain {}", chain_id))?;
+                tx_details.push((tx_id, chain_id));
+            }
+        }
+
+        for (tx_id, chain_id) in tx_details.iter() {
+            loop {
+                let (status_res, maybe_tx_hash, maybe_simulation_url) =
+                    client.get_relay_status(tx_id).await?;
+
+                match status_res.status() {
+                    TransactionStatus::TransactionFinalized => {
+                        println!(
+                            "Relaying to chain {} succeeded with tx hash: {:?}",
+                            chain_id,
+                            maybe_tx_hash.unwrap_or("None".to_string())
+                        );
+                        break;
+                    }
+                    TransactionStatus::TransactionFailed
+                    | TransactionStatus::TransactionTimedout => {
+                        return Err(anyhow::anyhow!(
+                            "Relaying to chain {} failed with tx hash: {:?}, simulation url: {:?}",
+                            chain_id,
+                            maybe_tx_hash.unwrap_or("None".to_string()),
+                            maybe_simulation_url.unwrap_or("None".to_string())
+                        ));
+                    }
+                    _ => {
+                        std::thread::sleep(Duration::from_secs(5));
+                    }
+                }
+            }
+        }
+
+        Ok(tx_details.into_iter().map(|(tx_id, _)| tx_id).collect())
+    }
+
+    pub async fn relay_proof_if_required(proof_id: String) -> Result<()> {
+        if let std::result::Result::Ok(chains_env) = env::var("CHAINS") {
+            if !chains_env.is_empty() {
+                log::info!("CHAINS is set, relaying proofs");
+                let access_token = env::var("PROVER_NETWORK_ACCESS_TOKEN")?;
+
+                let chain_ids: Vec<u32> = Self::parse_env_var_array("CHAINS")?;
+                let callbacks: Vec<String> = Self::parse_env_var_array("CALLBACKS")?;
+                let callback_datas: Vec<String> = Self::parse_env_var_array("CALLBACK_DATAS")?;
+
+                if !(chain_ids.len() == callbacks.len() && callbacks.len() == callback_datas.len())
+                {
+                    anyhow::bail!("CHAINS, CALLBACKS, and CALLBACK_DATAS must be of the same size");
+                }
```

Comment:
I don't think it makes sense to configure all of this through env

Reply:
Yeah I agree, in a subsequent PR I can add functions where you just pass in stuff like this so that an operator can use this lib. 

I don't know the best design for those interfaces yet (probably will start with tendermint) so this PR will just focus on the base mechanics.